Testscripts/Linux/channel_change.sh

```diff
@@ -0,0 +1,181 @@
+
+				# Read Rel_ID and target_cpu lines
+				read -a line
+				while [[ $line != '' ]]
+				do
+					rel_id=${line[0]#"$prefix1"}
+					rel_id=${rel_id%"$suffix1"}
+					LogMsg "Found Rel_ID: $rel_id"
+					cpu_id=${line[1]#"$prefix2"}
+					LogMsg "Found target_cpu: $cpu_id"
+					if [[ $cpu_id != "0" ]]; then
+						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
+						idle_cpus+=($_cpu_id)
+						LogMsg "Found the cpu id, $_cpu_id from the channel $rel_id. Will set 0, the default cpu id, to this cpu"
+						echo 0 > $sysfs_path/channels/$rel_id/cpu
+						sleep 1
+						# validate the change of that cpu id
+						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
+						if [[ $_cpu_id == "0" ]]; then
+							LogMsg "Successfully changed the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0"
+							cpu_idx=$((cpu_idx+1))
+						else
+							LogErr "Failed to change the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0. Expected 0, but found $_cpu_id"
+						fi
+					fi
+					read -a line
+				done
+				#LogMsg "Found vmbus channel and its target cpus: ${temp[@]}, ${!temp[@]}"
+			else
+				LogErr "Supposed to read Sysfs path, but found ${line[@]}"
+			fi
+			idx=$((idx+1))
+		else
+			LogErr "Supposed to read VMBUS ID line, but found ${line[@]}"
+		fi
+		LogMsg ""
+	done < "$lsvmbus_output_location"
+
+	LogMsg "VMBUS scanning result: ${vmbus_id[@]}"
+	return
+}
+
+function Main() {
+	# Collect the vmbus and cpu id information from the system, and store in /tmp/lsvmbus.output
+	basedir=$(pwd)
+	lsvmbus
+	if [ $? != 0 ]; then
+		if [ -f $basedir/linux/tools/hv/lsvmbus ]; then
+			chmod +x $basedir/linux/tools/hv/lsvmbus
+			$basedir/linux/tools/hv/lsvmbus -vv > $lsvmbus_output_location
+		else
+			LogErr "File, lsvmbus, not found in the system. Aborted the execution."
+			SetTestStateAborted
+			exit 0
+		fi
+	else
+		lsvmbus -vv > $lsvmbus_output_location
+	fi
+	LogMsg "Successfully recorded lsvmbus output in $lsvmbus_output_location"
+
+	# Before chaning cpu id to 0, reset all cpu ids of each vmbus channel.
+	LogMsg "Resetting CPU ID if non-zero"
+	reset_cpu_id
+
+	# Select a CPU number where does not associate to vmbus channels from idle_cpus array
+	for id in ${idle_cpus[@]}; do
+		state=$(cat /sys/devices/system/cpu/cpu$id/online)
+		if [[ $state == "1" ]]; then
+			LogMsg "Verified the current cpu $id is online"
+			dmesg > /tmp/pre-stage.log
+			LogMsg "Took a snapshot of dmesg to /tmp/pre-stage.log"
+			# set cpu to offline
+			echo 0 > /sys/devices/system/cpu/cpu$id/online
+			LogMsg "Changed the cpu $id state to offline"
+			sleep 1
+			post_state=$(cat /sys/devices/system/cpu/cpu$id/online)
+			if [[ $post_state == "0" ]]; then
+				LogMsg "Successfully verified to change the cpu $id state to offline"
+				sleep 1
+				dmesg > /tmp/post-stage.log
+				diff_val=$(diff /tmp/pre-stage.log /tmp/post-stage.log)
+				if [[ $diff_val == *"CPU $id is now offline"* ]]; then
```

Comment:
I think LISAv2 capture the system log or dmesg inside of Test log package. I am not sure if it is good to flush out.

Reply:
Ok. I think it would be good to use ```sysfs``` entry to check cpu is offline or online. Not relying on dmesg logs.
```/sys/devices/system/cpu/online``` & ```/sys/devices/system/cpu/offline```

Reply:
dmeg log check is a requirement.